Appraisals

```diff
@@ -65,1595 +58,14 @@ def self.gem_cucumber(version)
-    gem 'active_model_serializers', '>= 0.10.0'
-    gem 'activerecord', '~> 7'
-    gem 'aws-sdk'
-    gem 'concurrent-ruby'
-    gem 'dalli', '>= 3.0.0'
-    gem 'delayed_job'
-    gem 'delayed_job_active_record'
-    gem 'elasticsearch', '>= 8.0.0'
-    gem 'ethon'
-    gem 'excon'
-    gem 'grape'
-    gem 'graphql', '>= 2.0'
-    gem 'grpc', '>= 1.38.0', platform: :ruby # Minimum version with Ruby 3.0 support
-    gem 'http'
-    gem 'httpclient'
-    gem 'lograge'
-    gem 'makara', '>= 0.6.0.pre' # Ruby 3 requires >= 0.6.0, which is currently in pre-release: https://rubygems.org/gems/makara/versions
-    gem 'minitest', '>= 5.0.0'
-    gem 'mongo', '>= 2.8.0', '< 2.15.0' # TODO: FIX TEST BREAKAGES ON >= 2.15 https://github.com/DataDog/dd-trace-rb/issues/1596
-    gem 'mysql2', '>= 0.5.3', platform: :ruby
-    gem 'activerecord-jdbcmysql-adapter', platform: :jruby
-    gem 'opensearch-ruby'
-    gem 'pg', platform: :ruby
-    gem 'activerecord-jdbcpostgresql-adapter', platform: :jruby
-    gem 'racecar', '>= 0.3.5'
-    gem 'rack'
-    gem 'rack-contrib'
-    gem 'rack-test'
-    gem 'rake', '>= 12.3'
-    gem 'rest-client'
-    gem 'resque'
-    gem 'roda', '>= 2.0.0'
-    gem 'ruby-kafka', '>= 0.7.10'
-    gem 'rspec', '>= 3.0.0'
-    gem 'semantic_logger', '~> 4.0'
-    gem 'sequel', '~> 5.54.0' # TODO: Support sequel 5.62.0+
-    gem 'shoryuken'
-    gem 'sidekiq', '~> 7'
-    gem 'sneakers', '>= 2.12.0'
-    gem 'sqlite3', '>= 1.4.2', platform: :ruby
-    gem 'stripe'
-    gem 'jdbc-sqlite3', '>= 3.28', platform: :jruby
-    gem 'sucker_punch'
-    gem 'typhoeus'
-    gem 'que', '>= 1.0.0'
-    gem 'net-smtp'
-  end
-
-  [3, 4, 5].each do |n|
-    appraise "redis-#{n}" do
-      gem 'redis', "~> #{n}"
-    end
-  end
-
-  appraise 'sinatra' do
-    gem 'sinatra', '>= 3'
-    gem 'rack-test'
-  end
-
-  appraise 'opentelemetry' do
-    gem 'opentelemetry-sdk', '~> 1.1'
-  end
-
-  appraise 'contrib-old' do
-    gem 'dalli', '< 3.0.0'
-    gem 'elasticsearch', '< 8.0.0' # Dependency elasticsearch-transport renamed to elastic-transport in >= 8.0
-    gem 'graphql', '~> 1.12.0', '< 2.0' # TODO: Support graphql 1.13.x
-    gem 'presto-client', '>= 0.5.14' # Renamed to trino-client in >= 1.0
-
-    if RUBY_PLATFORM == 'java'
-      gem 'qless', '0.10.0' # Newer releases require `rusage`, which is not available for JRuby
-      gem 'redis', '< 4' # Missing redis version cap for `qless`
-    else
-      gem 'qless', '0.12.0'
-    end
-  end
-
-  appraise 'core-old' do
-    gem 'dogstatsd-ruby', '~> 4'
-  end
-  # ----------------------------------------------------------------------------------------------------------------------
-  # ADD NEW RUBIES HERE
-end
-
 ruby_runtime = if defined?(RUBY_ENGINE_VERSION)
                  "#{RUBY_ENGINE}-#{RUBY_ENGINE_VERSION}"
                else
                  "#{RUBY_ENGINE}-#{RUBY_VERSION}" # For Ruby < 2.3
                end
 
+instance_eval IO.read("appraisal/#{ruby_runtime}.rb")
```

Comment:
Have you considered not including the Ruby minor version in the file names? E.g. Ruby 3.2 instead of 3.2.0?

That would allow us to easily bump Ruby versions we use for CI testing without having to do big mass renames (I know appraisals already does this crime, but ahaha since we're adding this new one, we can get it right from the start ;)

Reply:
That seems reasonable. I think I will take one small step at a time.